src/framework/MOM_diag_mediator.F90

```diff
@@ -219,6 +244,25 @@ subroutine set_axes_info(G, param_file, diag_cs, set_vertical)
     id_zl = -1 ; id_zi = -1
   endif
 
+  ! Read info needed for z-space remapping
+  call get_param(param_file, mod, "Z_OUTPUT_GRID_FILE", zgrid_file, &
+                 "The file that specifies the vertical grid for \n"//&
+                 "depth-space diagnostics, or blank to disable \n"//&
+                 "depth-space output.", default="")
+  if (len_trim(zgrid_file) > 0) then
+    call get_param(param_file, mod, "INPUTDIR", in_dir, &
+                 "The directory in which input files are found.", default=".")
+    in_dir = slasher(in_dir)
+    call get_Z_output_grid(trim(in_dir)//trim(zgrid_file), "zw", diag_cs%z_remap_int, &
+                           "zt", id_zzl, id_zzi, diag_cs%nz_remap)
+    call log_param(param_file, mod, "!INPUTDIR/Z_OUTPUT_GRID_FILE", &
+                   trim(in_dir)//trim(zgrid_file))
+    call log_param(param_file, mod, "!NK_ZSPACE (from file)", diag_cs%nz_remap, &
+                 "The number of depth-space levels.  This is determined \n"//&
+                 "from the size of the variable zw in the output grid file.", &
+                 units="nondim")
+  endif
```

Comment:
Can we introduce a more succinct syntax such used for "ALE_COORDINATE_CONFIG?
e.g.
ocean_model_new_z,FILE:vgrid.nc,zt
and use "call MOM_read_data(trim(fileName), trim(varName), dz)"